crates/dbsp/src/trace/ord/fallback/indexed_wset.rs

```diff
@@ -393,11 +393,17 @@ where
         Self::Builder::new_builder(factories, time).done()
     }
 
-    fn persist(&mut self) {
-        if let Inner::Vec(vec) = &self.inner {
-            let mut file = FileIndexedWSetBuilder::with_capacity(&self.factories.file, (), 0);
-            copy_to_builder(&mut file, vec.cursor());
-            self.inner = Inner::File(file.done());
+    fn persisted(&self) -> Option<Self> {
+        match &self.inner {
+            Inner::Vec(vec) => {
+                let mut file = FileIndexedWSetBuilder::with_capacity(&self.factories.file, (), 0);
+                copy_to_builder(&mut file, vec.cursor());
```

Comment:
could this take a long time?
is that not a problem?

Reply:
It could take a long time. I don't know a way to write data to storage that won't.

Reply:
the question was more about needing an async API

Reply:
An async API wouldn't help anything in this case because we're committing synchronously. I guess if we had an async commit API then async write-to-disk might help.